### PR TITLE
search vendor and server side pagination

### PR DESCRIPTION
**Pull request by Sree0076**
added server side pagination and search in vendor table

### CODE REVIEW

File: src/app/features/vendor/vendor-listing/vendor-listing.component.html [hunk 1]
```diff
@@ -13,13 +13,34 @@ <h2>All Vendors</h2>
...
+      [responsive]="true"
     >
+    <ng-template pTemplate="caption">
+      <div class="flex">
+          <p-iconField iconPosition="left" class="ml-auto">
+              <p-inputIcon>
+                  <i class="pi pi-search"></i>
+              </p-inputIcon>
+              <input 
+                pInputText
```
Comment (kavyamolr) [inline src/app/features/vendor/vendor-listing/vendor-listing.component.html:37]: id missing

File: src/app/features/vendor/vendor-listing/vendor-listing.component.ts [hunk 2]
```diff
@@ -36,15 +39,23 @@ export class VendorListingComponent implements OnInit {
 
   ngOnInit() {
     this.loadingService.showLoader();
-    this.fetchVendors();
   }
 
+  onPageChange(first?: number, rows?: number | null): void {
```
Comment (kavyamolr) [inline src/app/features/vendor/vendor-listing/vendor-listing.component.ts:50]: method description missing

File: src/app/features/vendor/vendor-listing/vendor-listing.component.ts [hunk 3]
```diff
@@ -55,4 +66,13 @@ export class VendorListingComponent implements OnInit {
   navigateToView(event: IVendor): void {
     this.router.navigate(['/vendor/view/' + event.id]);
   }
+
+  /**
+   * Method to search a vendor by name
+   */
+  filterGlobal(event: Event) {
```
Comment (kavyamolr) [inline src/app/features/vendor/vendor-listing/vendor-listing.component.ts]: add meaningful name

File: src/app/features/vendor/vendor-listing/vendor-listing.component.html [hunk 4]
```diff
@@ -13,13 +13,35 @@ <h2>All Vendors</h2>
...
     >
+    <ng-template pTemplate="caption">
+      <div class="flex">
+          <p-iconField iconPosition="left" class="ml-auto">
+              <p-inputIcon>
+                  <i class="pi pi-search"></i>
+              </p-inputIcon>
+              <input 
+                pInputText 
+                id="search"
```
Comment (kavyamolr) [inline src/app/features/vendor/vendor-listing/vendor-listing.component.html]: change to search-vendor-input

File: src/app/shared/breadcrumb/breadcrumb.component.ts [hunk 5]
```diff
@@ -58,7 +58,7 @@ export class BreadcrumbComponent implements OnInit {
    */
   buildBreadcrumb(route: ActivatedRoute): MenuItem[] {
     const children: ActivatedRoute[] = route.children;
-
+    console.log(children)
```
Comment (kavyamolr) [inline src/app/shared/breadcrumb/breadcrumb.component.ts]: remove console

File: src/app/features/vendor/vendor-listing/vendor-listing.component.html (hunk 1 above)
Comment (Sree0076) [reply src/app/features/vendor/vendor-listing/vendor-listing.component.html:37]: done

File: src/app/features/vendor/vendor-listing/vendor-listing.component.ts (hunk 3 above)
Comment (Sree0076) [reply src/app/features/vendor/vendor-listing/vendor-listing.component.ts]: changed

File: src/app/features/vendor/vendor-listing/vendor-listing.component.html (hunk 4 above)
Comment (Sree0076) [reply src/app/features/vendor/vendor-listing/vendor-listing.component.html]: done

File: src/app/shared/breadcrumb/breadcrumb.component.ts (hunk 5 above)
Comment (Sree0076) [reply src/app/shared/breadcrumb/breadcrumb.component.ts]: done

**Review by kavyamolr — APPROVED** [review]

**Review by athul-sanjeevan — APPROVED** [review]

**Pull request merged**